### PR TITLE
fix(Consumer): dont publish to failures topic if not listening to failures

### PR DESCRIPTION
**Pull request by omamiya**
…

### CODE REVIEW

File: lib/yotpo_kafka/consumer.rb [hunk 1]
```diff
@@ -14,14 +14,19 @@ def initialize(params = {})
       @seed_brokers = params[:broker_url] || ENV['BROKER_URL'] || '127.0.0.1:9092'
       @kafka = Kafka.new(@seed_brokers)
       @seconds_between_retries = params[:seconds_between_retries] || 0
-      @listen_to_failures = true
-      @listen_to_failures = params[:listen_to_failures] unless params[:listen_to_failures].nil?
+
+      @json_parse = params[:json_parse].nil? ? true : params[:json_parse]
+      if !YotpoKafka.kafka_v2 && !@json_parse
+        @listen_to_failures = false
```
Comment (bronhaim) [inline lib/yotpo_kafka/consumer.rb]: add big log here

File: lib/yotpo_kafka/consumer.rb [hunk 2]
```diff
@@ -97,19 +102,12 @@ def consume_kafka_v2(payload, message)
...
+      handle_error_kafka_v2(message, error, print_payload)
     rescue SignalException => error
-      log_error('Signal Exception sending message to retry in kafka v2 and closing server',
+      log_error('Signal Exception',
                 error: error.message,
-                topic: message.topic,
-                payload: print_payload,
                 backtrace: error.backtrace)
-      handle_error_kafka_v2(message, error)
+      handle_error_kafka_v2(message, error, print_payload)
```
Comment (bronhaim) [inline lib/yotpo_kafka/consumer.rb]: rename print_payload to encoded_payload_for_print or something. the name is too confusing

File: lib/yotpo_kafka/consumer.rb [hunk 3]
```diff
@@ -97,19 +102,12 @@ def consume_kafka_v2(payload, message)
                 broker_url: @seed_brokers)
       consume_message(payload)
     rescue => error
-      log_error('consume_kafka_v2 failed in service - handling retry',
-                error: error.message,
-                topic: message.topic,
-                payload: print_payload,
-                backtrace: error.backtrace)
-      handle_error_kafka_v2(message, error)
+      handle_error_kafka_v2(message, error, print_payload)
```
Comment (bronhaim) [inline lib/yotpo_kafka/consumer.rb]: why the if @listen_to_failures is inside the function. do the if here no?

File: yotpo_kafka.gemspec [hunk 4]
```diff
@@ -3,7 +3,7 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 
 Gem::Specification.new do |s|
   s.name = 'yotpo-ruby-kafka'
-  s.version = '1.0.43'
+  s.version = '1.0.44'
```
Comment (bronhaim) [inline yotpo_kafka.gemspec]: no tests? you're playing with fire ...

File: yotpo_kafka.gemspec [hunk 5]
```diff
@@ -3,7 +3,7 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
...
   s.date = '2019-11-13'
   s.authors = 'Dana & Yaniv'
```
Comment (bronhaim) [inline yotpo_kafka.gemspec]: it's time to rename it to GOPHERS

File: yotpo_kafka.gemspec [hunk 6]
```diff
@@ -3,7 +3,7 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 
 Gem::Specification.new do |s|
   s.name = 'yotpo-ruby-kafka'
-  s.version = '1.0.43'
+  s.version = '1.0.44'
   s.date = '2019-11-13'
```
Comment (bronhaim) [inline yotpo_kafka.gemspec]: update dates - it will be easier for you to track changes

File: lib/yotpo_kafka/consumer.rb [hunk 7]
```diff
@@ -97,19 +102,12 @@ def consume_kafka_v2(payload, message)
                 broker_url: @seed_brokers)
```
Comment (bronhaim) [inline lib/yotpo_kafka/consumer.rb]: what about printing the partition and offset and change this print to something more interesting than consumer failed to start?

Comment (omamiya) [reply lib/yotpo_kafka/consumer.rb]: i think it's informative enough

File: lib/yotpo_kafka/consumer.rb (hunk 3 above)
Comment (omamiya) [reply lib/yotpo_kafka/consumer.rb]: then i need to do it twice. here and in the second place we call handle_error_kafka_v* (after catch the SignalException)

File: lib/yotpo_kafka/consumer.rb (hunk 2 above)
Comment (omamiya) [reply lib/yotpo_kafka/consumer.rb]: dont think so...

File: yotpo_kafka.gemspec (hunk 6 above)
Comment (omamiya) [reply yotpo_kafka.gemspec]: 👍

File: yotpo_kafka.gemspec (hunk 4 above)
Comment (omamiya) [reply yotpo_kafka.gemspec]: צודק

File: yotpo_kafka.gemspec (hunk 5 above)
Comment (omamiya) [reply yotpo_kafka.gemspec]: yup

File: lib/yotpo_kafka/consumer.rb (hunk 1 above)
Comment (omamiya) [reply lib/yotpo_kafka/consumer.rb]: done

**Pull request merged**